agent/stats/engine.go

```diff
@@ -438,16 +440,21 @@ func (engine *DockerStatsEngine) GetInstanceMetrics(includeServiceConnectStats b
 		}
 	}
 
-	for taskArn := range engine.tasksToContainers {
-		containerMetrics, err := engine.taskContainerMetricsUnsafe(taskArn)
-		if err != nil {
-			seelog.Debugf("Error getting container metrics for task: %s, err: %v", taskArn, err)
-			continue
-		}
+	taskStatsToCollect := engine.getTaskStatsToCollect()
+	for taskArn := range taskStatsToCollect {
+		containerMetrics := make([]*ecstcs.ContainerMetric, 0)
+		var err error
+		if !idle {
+			containerMetrics, err = engine.taskContainerMetricsUnsafe(taskArn)
+			if err != nil {
+				seelog.Debugf("Error getting container metrics for task: %s, err: %v", taskArn, err)
+				continue
```

Comment:
If there is no container metrics for this task, it will not collect service-connect metrics also for that task, right?

Reply:
right but if that is task is SC enabled then `engine.taskToServiceConnectStats` will not be empty and we will still collect SC metrics for that task.

Reply:
Addressed it in the latest commit, thank you for calling it.